src/types/local_date.rs

```diff
@@ -0,0 +1,268 @@
+use crate::utils::js_error;
+use scylla::frame::value::CqlDate;
+use std::{cmp::max, fmt, num::ParseIntError};
+
+// max and min date range of the Date class in JS.
+const MAX_JS_DATE: i32 = 100_000_000;
+const MIN_JS_DATE: i32 = -MAX_JS_DATE;
+
+#[napi]
+pub struct LocalDateWrapper {
+    pub value: i32, // wrapper for number of days from 01.01.1970
+    pub year: i32,
+    pub month: i8,
+    pub day: i8,
+    pub in_date: bool, // value can be represented as Date class in JS
+}
+
+#[napi]
+impl LocalDateWrapper {
+    /// Create a new object from the day, month and year.
+    #[napi]
+    pub fn new(day: i8, month: i8, year: i32) -> napi::Result<Self> {
+        Self::check_valid_date(day, month, year)
+    }
+
+    /// Create a new object from number of days since 01.01.1970.
+    #[napi]
+    pub fn new_day(value: i32) -> napi::Result<Self> {
+        let mut x = LocalDateWrapper::new(1, 1, 1970)?;
+        x.value = value;
+        x.in_date = (MIN_JS_DATE..=MAX_JS_DATE).contains(&value);
+        if x.in_date {
+            x.add_day(value.into());
+        }
+        Ok(x)
+    }
+
+    /// Counts the number of days since 01.01.1970.
+    fn days_since_epoch(day: i8, month: i8, year: i32) -> i32 {
+        let mut total_days = 0;
+
+        if year >= 1970 {
+            for m in 1..month {
+                total_days += Self::days_in_month(m, year) as i32;
+            }
```

Comment:
Again: performance. Here we can maybe preprocess the values for each month accounting the leap year.